blueocean-dashboard/src/main/js/components/karaoke/components/LogConsole.jsx

```diff
@@ -13,33 +13,38 @@ const RERENDER_DELAY = 17;
 
 const logger = logging.logger('io.jenkins.blueocean.dashboard.karaoke.LogConsole');
 
-const LogLine = ({ prefix, line, index, router, location }) => {
-    const tokenized = tokenizeANSIString(line);
-    const lineChunks = makeReactChildren(tokenized);
+class LogLine extends PureComponent {
+    onClick = () => {
+        const { prefix, index, router, location } = this.props;
 
-    const onClick = () => {
         const loc2 = location; // For eslint 🙄
         loc2.hash = `#${prefix || ''}log-${index + 1}`;
         router.push(loc2);
     };
 
-    return (
-        <p key={index + 1} id={`${prefix}log-${index + 1}`}>
-            <div className="log-boxes">
-                <a
-                    className="linenumber"
-                    key={index + 1}
-                    href={`#${prefix || ''}log-${index + 1}`}
-                    name={`${prefix}log-${index + 1}`}
-                    onClick={onClick}
-                >
-                </a>
-                {
-                    React.createElement(Linkify, { className: 'line ansi-color' }, ...lineChunks)
-                }
-            </div>
-        </p>
-    );
+    render() {
+        const { prefix, line, index } = this.props;
+        const tokenized = tokenizeANSIString(line);
+        const lineChunks = makeReactChildren(tokenized);
+
+        return (
+            <p id={`${prefix}log-${index + 1}`}>
+                <div className="log-boxes">
+                    <a
+                        className="linenumber"
+                        key={index + 1}
+                        href={`#${prefix || ''}log-${index + 1}`}
+                        name={`${prefix}log-${index + 1}`}
+                        onClick={this.onClick}
```

Comment:
this needs to be something like `() => this.onClick()`

Reply:
Because we're using `PureComponent` now, rather than a stateless function, we can directly pass in `this.onClick` to the `a` react element without needing to wrap it in a arrow function. The functionality still works as expected, without the need for creating a new function on each render.

Let me know if I've mis-understood, or if it's the preferred code style of blueocean to wrap callbacks in arrow functions :+1:

Reply:
Pretty sure that `this` gets lost in the execution context when using ES6 unless you use `.bind(this)`, does it not?

Reply:
Check out line 17:

```js
onClick = () => {
```

It's doing binding for us :)

Reply:
ah heh, missed that 😬